content/docs/learning-guide/tutorials/rendering/vegetation-bending-tutorial.md

```diff
@@ -0,0 +1,834 @@
+
+   return OUT;
+   ``` 
+   {{< note >}}
+   We need the code to be above the two `OUT` lines because we will edit `worldPosition` and then will need to adjust the `OUT` variables accordingly.
+   {{< /note >}}
+  
+1. Repeat steps 2-3 with the depth pass in `VegetationBending_DepthPass.azsli` and the depth pass with PS in `VegetationBending_DepthPass_WithPS.azsli`.
+
+   {{< tip >}}
+   Note that the depth pass does not output a world position. Just place the code above the `OUT.m_position` line. The depth pass with PS has some extra code in the vertex shader, but you can still place this code about the two `OUT` statements.
+   {{< /tip >}}
+
+### Set up wind bending
+Let's begin editing the code to add wind. 
+
+1. Open `VegetationBending_Common.azsli`.
+
+1. At the bottom, add a function to calculate the amplitude, frequency, and phase of the wind according to the time and world position of the vertex. The phase should take into account the `worldPosition` because we want to mimic how wind should affect nearby objects similarly but faraway objects differently. Faraway objects may not be affected by the same "breeze". This function will then later be used to calculate the appropriate movement of the vertex.
+
+   ```hlsl
+   float4 SetUpWindBending(float currentTime, float4 worldPosition) 
+   {
+      float2 wind = float2(MaterialSrg::m_windX, MaterialSrg::m_windY);
+      float bendingStrength = MaterialSrg::m_bendingStrength;
+      float2 amplitude = float2(wind.x * 0.4 + wind.y * 0.2, wind.y * 0.4 - wind.x * 0.2);
+      float2 frequency = float2(MaterialSrg::m_windBendingFrequency, MaterialSrg::m_windBendingFrequency * 1.125);
+      // Using the world position to modify the phase makes it so different trees near each other are at similar but not equal points in the animation, 
+      // so they appear to be reacting to the same wind but at different times as the wind moves through the vegetation.
+      float2 phase = worldPosition.xy * 0.08;
+
+      float2 bendAmount = sin(currentTime * frequency + phase) * amplitude;
+
+      float4 result;
+      result.xy = bendAmount + wind;
+      result.z = length(wind);
+      result.w = 0.3 * length(result.xy);
+      result.xyz *= bendingStrength * 0.08;
+
+      return result;
+   }
+   ```
+
+   {{< note >}}
+   This function could potentially be run once per-object on the CPU with the results updated each frame in the ObjectSrg instead of being run once per-vertex on the GPU. It's a tradeoff between doing the computation once per frame per vegetation object + an extra SRG compile per frame per vegetation object vs. re-doing the computation on the GPU per frame per vegetation object per vertex. It's content specific, with the redundant GPU cost increasing as vertex density increases, and also depends on if the GPU is the bottleneck, and whether or not the vertex shader is the bottleneck, and also if the vertex shader is bandwidth bound or ALU bound. 
```

Comment:
Can you do the following:
- expand the full term for ALU, then put ALU in parenthesis
- break apart the long sentences to avoid run-ons
- break apart into shorter paragraphs

```suggestion
   This function could potentially be run once per-object on the CPU with the results updated each frame in the ObjectSrg instead of being run once per-vertex on the GPU. It's a tradeoff between doing the computation once per frame per vegetation object, plus an extra SRG compile per frame per vegetation object versus re-doing the computation on the GPU per frame per vegetation object per vertex. It's content specific, with the redundant GPU cost increasing as vertex density increases, and also depends on if the GPU is the bottleneck, and whether or not the vertex shader is the bottleneck, and also if the vertex shader is bandwidth bound or ALU bound. 
```